### ISSUE TITLE
Use `crypto.webcrypto.*` functions to be compliant with the Web Crypto spec

### ISSUE THREAD

**Issue by MichaelDeBoey** [tj/node-cookie-signature#49]
https://nodejs.org/dist/latest-v20.x/docs/api/crypto.html#cryptosubtle

CC/ @natevw

**Comment by MichaelDeBoey** [comment]
@natevw My goal was to make this a first step towards full Web Crypto compliance (using the `crypto` global instead of the Crypto API), so that this package could be used on other runtimes like Cloudflare and/or Deno as well
- https://developers.cloudflare.com/workers/runtime-apis/web-crypto
- https://deno.land/api@v1.36.2?s=SubtleCrypto

**Comment by natevw** [comment]
Yeah, that does seem useful at least in some library. But the code here is basically two tiny functions that have worked ± the same for over a decade.

@dougwilson is the Express framework moving to support other runtimes besides Node.js? Otherwise I'm inclined to leave this work available to a new npm package that can have a fresh start in the modern JS era.

**Comment by MichaelDeBoey** [comment]
@natevw The end-result of these changes will be exactly the same as it was without these change, it's only an internal change of using different APIs
But it has the added benefit of being usable with Bun, Cloudflare and Deno as well

**Comment by nexdrew** [comment]
If anyone's interested, I recently published a new package called [`cookie-signature-subtle`](https://www.npmjs.com/package/cookie-signature-subtle) that does exactly what @MichaelDeBoey is asking for - it's basically `cookie-signature` using the Web Crypto API (which is async) - because I had a similar desire to support other non-Node web-standard runtimes. (Apologies for the shameless plug, just trying to help!)

**Comment by natevw** [comment]
@nexdrew No apologies needed, thanks for sharing! Seems to match exactly my hope that someone could make a fresh start without the backwards compatibility constraints. If more people end up needing this we could probably update the README as well (and perhaps there's other similar libraries in the space…).
